### PR TITLE
Vectra Detect Analytics fixes

### PR DESCRIPTION
**Pull request by danymello**
Required items, please complete
   
   Change(s):
   - fix a couple of issues that I did not capture and automated validation neither
   - Add alertDetailsOverride 
   - Add incidentConfiguration for relevant template
   - All templates are now set to Informational and has dynamic Severity mapped from Vectra
   - Add custom details

   Reason for Change(s):
   - Missing incident creation configuration
   - Missing override for alerts
   - Wrong entity mapping
   - Wrong default severity
  
   Version Updated:
   - yes

   Testing Completed:
   - yes.

   Checked that the validations are passing and have addressed any issues that are present:
   - tbd

### CODE REVIEW

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ query: |
   | extend Severity = case(level == "Info", "Informational",
       level == "Low", "Low",
       level == "Medium", "Medium",
-      level == "High", "Medium",
+      level == "High", "High",
       level == "Critical", "High",
```
Comment (devikamehra) [inline Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml:63]: Entire block of this case statement can be improvement by using the following:
case(level == "Info", "Informational",
      level == "Critical", "High",
      level)

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml [hunk 2]
```diff
@@ -71,9 +71,9 @@ entityMappings:
         columnName: name
       - identifier: UPNSuffix
         columnName: upn_suffix
-alertDetails:
-  alertDisplayNameFormat: Vectra AI - {{Activity}} Detected
-  alertDescriptionFormat: >-
+alertDetailsOverride:
+  alertDisplayNameFormat: "Vectra AI - {{Activity}} Detected"
```
Comment (devikamehra) [inline Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml:75]: Additional quotes are not required in alertDisplayNameFormat or alertDescriptionFormat.
Applicable to all the queries.

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-HighSeverityDetection-by-Tactics.yaml [hunk 3]
```diff
@@ -85,5 +98,15 @@ alertDetailsOverride:
...
+  createIncident: true
+  groupingConfiguration:
+    enabled: true
+    reopenClosedIncident: true
+    lookbackDuration: 7d
+    matchingMethod: AllEntities
+customDetails:
+  AttackType: Activity
+  AttackCategory: Category  
 version: 1.0.5
```
Comment (devikamehra) [inline Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-HighSeverityDetection-by-Tactics.yaml:111]: Version not updated

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml [hunk 4]
```diff
@@ -39,6 +39,7 @@ query: |
   | project-rename threat_score = FlexNumber1
   | project-rename certainty_score = FlexNumber2
   | project-rename triaged = DeviceCustomString5
+  | where triaged != "True" and AdditionalExtensions !contains "account"
```
Comment (devikamehra) [inline Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml:42]: Can this "contains" be replaced with "has"?

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml [hunk 5]
```diff
@@ -89,5 +92,8 @@ alertDetailsOverride:
       value: DeviceVendor
     - alertProperty: ConfidenceScore
       value: certainty_score
+customDetails:
+  AttackType: Activity
+  AttackCategory: Category
 version: 1.0.6
```
Comment (devikamehra) [inline Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml:98]: Version not updated

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml (hunk 2 above)
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml:75]: good to know @devikamehra

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml (hunk 5 above)
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml]: changed

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml (hunk 4 above)
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Suspected-Behavior-by-Tactics.yaml]: done in all templates

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-HighSeverityDetection-by-Tactics.yaml (hunk 3 above)
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-HighSeverityDetection-by-Tactics.yaml]: thanks

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml (hunk 1 above)
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml]: implemented

**Comment by danymello** [comment]
@devikamehra. I implemented all the changes.

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml (hunk 2 above)
Comment (devikamehra) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-Account-Detections.yaml:70]: Can we remove these?

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-NewCampaign.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@ id: a34d0338-eda0-42b5-8b93-32aae0d7a501
 name: Vectra AI Detect - New Campaign Detected
 description: |
   'Identifies when a new Campaign has been detected.  This occurs when multiple Detections accross different Hosts are suspected to be part of the same Attack Campaign.'
-severity: Medium
+severity: Informational
```
[the remark this answers is not in the document]
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-NewCampaign.yaml]: changed.

File: Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-HighSeverityDetection-by-Tactics.yaml [hunk 7]
```diff
@@ -4,14 +4,14 @@ description: |
   'Create an incident for high severity malicious behavior detected by Vectra AI (Threat score superior to 7.0). 
   The Severity is a mapping with the Threat score assigned to a detection. It ranges between 0 and 10. 
   The severity_threshold variable can be adjusted as desired.' 
-severity: High
+severity: Informational
```
[the remark this answers is not in the document]
Comment (danymello) [reply Solutions/Vectra AI Detect/Analytic Rules/VectraDetect-HighSeverityDetection-by-Tactics.yaml]: correct. I used dynamic mapping but I can set it to high directly

**Comment by danymello** [comment]
@devikamehra , all done! ty

**Comment by danymello** [comment]
> @danymello The extra quotes are not removed. Can you please make those changes aswell? @v-atulyadav I observed that DeviceVendor is mapped to both ProviderName and ProductName. Can we map DeviceProduct to ProductName as part of this PR itself. This will allow us to package this once.

@devikamehra I removed the quotes. 
@v-atulyadav, the "DeviceProduct" attribute does not make a lot of sense in our CEF format. It refers to a type of appliance but it does not exist anymore. It makes more sense to keep DeviceVendor mapped to DeviceVendor as well.

**Comment by devikamehra** [comment]
> > @danymello The extra quotes are not removed. Can you please make those changes aswell? @v-atulyadav I observed that DeviceVendor is mapped to both ProviderName and ProductName. Can we map DeviceProduct to ProductName as part of this PR itself. This will allow us to package this once.
> 
> @devikamehra I removed the quotes. @v-atulyadav, the "DeviceProduct" attribute does not make a lot of sense in our CEF format. It refers to a type of appliance but it does not exist anymore. It makes more sense to keep DeviceVendor mapped to DeviceVendor as well.

@danymello Thanks for making this change. Can you please perform the same in VectraDetect-HighSeverityDetection-by-Tactics.yaml as well.

**Comment by danymello** [comment]
> > > @danymello The extra quotes are not removed. Can you please make those changes aswell? @v-atulyadav I observed that DeviceVendor is mapped to both ProviderName and ProductName. Can we map DeviceProduct to ProductName as part of this PR itself. This will allow us to package this once.
> > 
> > 
> > @devikamehra I removed the quotes. @v-atulyadav, the "DeviceProduct" attribute does not make a lot of sense in our CEF format. It refers to a type of appliance but it does not exist anymore. It makes more sense to keep DeviceVendor mapped to DeviceVendor as well.
> 
> @danymello Thanks for making this change. Can you please perform the same in VectraDetect-HighSeverityDetection-by-Tactics.yaml as well.

@devikamehra - it is done!

**Comment by devikamehra** [comment]
@danymello Thank you for mapping all the changes. I notice that this PR is raised through your forked branch. Can you please help us fix the DeviceProduct field as well. In case it is not possible because of your tight schedule, I will understand and ask my team to raise a separate PR for this and approve this one.
Please let me know your thoughts.

**Comment by devikamehra** [comment]
> > @danymello Thank you for mapping all the changes. I notice that this PR is raised through your forked branch. Can you please help us fix the DeviceProduct field as well. In case it is not possible because of your tight schedule, I will understand and ask my team to raise a separate PR for this and approve this one. Please let me know your thoughts.
> 
> @devikamehra - all updated!

Thank you so much @danymello for helping with this.

**Review by devikamehra — APPROVED** [review]

**Comment by devikamehra** [comment]
@v-atulyadav please review the deployment and merge. Code change looks good to me.

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**